crates/subspace-networking/src/create.rs

```diff
@@ -170,14 +178,29 @@ pub async fn create(config: Config) -> Result<(Node, NodeRunner), CreationError>
         pieces_by_range_request_handler,
         relay_server_address,
         parent_node,
+        networking_parameters_registry,
     } = config;
     let local_peer_id = keypair.public().to_peer_id();
-
     // Create relay client transport and client.
     let (relay_transport, relay_client) = RelayClient::new_transport_and_behaviour(local_peer_id);
 
     let transport = build_transport(&keypair, timeout, yamux_config, relay_transport).await?;
 
+    let cached_bootstrap_addresses = networking_parameters_registry
+        .known_addresses(INITIAL_BOOTSTRAP_ADDRESS_NUMBER)
+        .await
+        .iter()
+        .cloned()
+        .filter(|(_, addr)| {
+            // filter Memory addresses
+            !addr
+                .into_iter()
+                .any(|protocol| matches!(protocol, Protocol::Memory(..)))
```

Comment:
Why not? Also I think memory addresses shouldn't be added to registry in the first place.

Reply:
Will move the check to the saving point.